### PR TITLE
Use trace range for time graph view boundary validation before a fetch

### LINKED ISSUE #304: TspDataProvider@getData() uses the first time graph entry for boundary check
TspDataProvider@getData() uses the first time graph entry for boundary check, which can lead that certain time range are not drawn when the first entry end time is smaller than other entries.

### PR DESCRIPTION
**Pull request by bhufmann**
fixes #304

Signed-off-by: Bernd Hufmann <Bernd.Hufmann@ericsson.com>

### CODE REVIEW

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts [hunk 1]
```diff
@@ -39,11 +41,11 @@ export class TspDataProvider {
...
         }
 
-        this.totalRange = this.timeGraphEntries[0].end - this.timeGraphEntries[0].start; // 1332170682540133097 - starttime
+        this.totalRange = totalTimeRange.getEnd() - totalTimeRange.getstart();
         let fetchParameters = QueryHelper.selectionTimeQuery(QueryHelper.splitRangeIntoEqualParts(1332170682440133097, 1332170682540133097, 1120), ids);
         if (viewRange && resolution) {
-            const start = viewRange.start + this.timeGraphEntries[0].start;
-            const end = viewRange.end + this.timeGraphEntries[0].start;
+            const start = viewRange.start + totalTimeRange.getstart();
+            const end = viewRange.end + totalTimeRange.getEnd();
```
Comment (PatrickTasse) [inline packages/react-components/src/components/data-providers/tsp-data-provider.ts:48]: Should be totalTimeRange.getstart() also

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts [hunk 2]
```diff
@@ -39,11 +41,11 @@ export class TspDataProvider {
             };
         }
 
-        this.totalRange = this.timeGraphEntries[0].end - this.timeGraphEntries[0].start; // 1332170682540133097 - starttime
+        this.totalRange = totalTimeRange.getEnd() - totalTimeRange.getstart();
         let fetchParameters = QueryHelper.selectionTimeQuery(QueryHelper.splitRangeIntoEqualParts(1332170682440133097, 1332170682540133097, 1120), ids);
         if (viewRange && resolution) {
-            const start = viewRange.start + this.timeGraphEntries[0].start;
-            const end = viewRange.end + this.timeGraphEntries[0].start;
+            const start = viewRange.start + totalTimeRange.getstart();
```
Comment (PatrickTasse) [inline packages/react-components/src/components/data-providers/tsp-data-provider.ts:47]: I would suggest to swap the order in next two lines (totalTimeRange.getstart() + viewRange.start/end)

**Review by PatrickTasse — CHANGES_REQUESTED** [review]

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts (hunk 1 above)
Comment (bhufmann) [reply packages/react-components/src/components/data-providers/tsp-data-provider.ts:48]: good catch

File: packages/react-components/src/components/data-providers/tsp-data-provider.ts (hunk 2 above)
Comment (bhufmann) [reply packages/react-components/src/components/data-providers/tsp-data-provider.ts:47]: Will do

**Review by PatrickTasse — APPROVED** [review]

**Review by MatthewKhouzam — APPROVED** [review]
Makes sense, I tried with an experiment that had a year between both traces. :)

**Pull request merged**